src/rp/index.ts

```diff
@@ -225,94 +233,72 @@ export class RelyingPartySolution {
+
     const response = await this.appFetch(requestObj.payload.response_uri, {
       method: "POST",
       headers: {
         "Content-Type": "application/x-www-form-urlencoded",
       },
-      body: formBody.toString(),
+      body,
     });
 
     if (response.status === 200) {
-      return response.text();
+      return await response.text();
     }
 
     throw new IoWalletError(
-      `Unable to send Authorization Response. Response code: ${response.status}`
+      `Unable to send Authorization Response. Response: ${await response.text()} with code: ${
+        response.status
+      }`
     );
   }
 
   /**
-   * Select a public key from those provided by the RP.
-   * Keys with algorithm "RSA-OAEP-256" or "RSA-OAEP" are expected, the firsts to be preferred.
+   * Select a RSA public key from those provided by the RP to encrypt.
    *
    * @param entity The RP entity configuration
    * @returns A suitable public key with its compatible encryption algorithm
    * @throws {NoSuitableKeysFoundInEntityConfiguration} If entity do not contain any public key suitable for encrypting
    */
-  private choosePublicKeyToEncrypt(
-    entity: RpEntityConfiguration
-  ): (JWK & { alg: "RSA-OAEP-256" }) | (JWK & { alg: "RSA-OAEP" }) {
-    // Look for keys using "RSA-OAEP-256", and pick a random one
-    const [usingRsa256] = entity.payload.jwks.keys.filter(
-      <T>(k: T & { alg?: string }): k is T & { alg: "RSA-OAEP-256" } =>
-        typeof k.alg === "string" && k.alg === "RSA-OAEP-256"
-    );
+  private chooseRSAPublicKeyToEncrypt(entity: RpEntityConfiguration): JWK {
+    const [usingRsa256] =
+      entity.payload.metadata.wallet_relying_party.jwks.filter(
+        (jwk) => jwk.use === "enc" && jwk.kty === "RSA"
+      );
 
     if (usingRsa256) {
       return usingRsa256;
     }
 
-    // Look for keys using "RSA-OAEP", and pick a random one
-    const [usingRsa] = entity.payload.jwks.keys.filter(
-      <T>(k: T & { alg?: string }): k is T & { alg: "RSA-OAEP" } =>
-        typeof k.alg === "string" && k.alg === "RSA-OAEP"
-    );
-
-    if (usingRsa) {
-      return usingRsa;
-    }
-
     // No suitable key has been found
     throw new NoSuitableKeysFoundInEntityConfiguration(
       "Encrypt with RP public key"
     );
   }
 
-  private getEncryptionAlgByJwk({
-    alg,
-  }: (JWK & { alg: "RSA-OAEP-256" }) | (JWK & { alg: "RSA-OAEP" })):
-    | "A128CBC-HS256"
-    | "A256CBC-HS512" {
-    if (alg === "RSA-OAEP-256") return "A256CBC-HS512";
-    if (alg === "RSA-OAEP") return "A128CBC-HS256";
-
-    const _: never = alg;
-    throw new Error(`Invalid jwk algorithm: ${_}`);
-  }
-
   /**
    * Obtain the relying party entity configuration.
    */
   async getEntityConfiguration(): Promise<RpEntityConfiguration> {
-    const wellKnownUrl = new URL(
-      "/.well-known/openid-federation",
-      this.relyingPartyBaseUrl
-    ).href;
+    const wellKnownUrl =
+      this.relyingPartyBaseUrl + "/.well-known/openid-federation";
```

Comment:
```suggestion
      new URL("/.well-known/openid-federation", this.relyingPartyBaseUrl).href;
```

Reply:
Done in https://github.com/pagopa/io-react-native-wallet/pull/31/commits/dced748fd8a931965ce63ae62aa76fc76d422afa

Reply:
The base url must be taken as-is, at it may have _path_ part. `URL` class will use only the hostname

Example:
```ts
new URL("/foo/bar", "https://example.com/base/path").href
// https://example.com/foo/bar
```